### PR TITLE
add boot2 info to binary info (fix #68)

### LINKED ISSUE #68: Add boot stage2 information to binary info
Would be nice for picotool to show this info too with -a

### PR DESCRIPTION
**Pull request by kilograham**

### CODE REVIEW

**Comment by lurch** [comment]
At first I thought this wasn't working, but then I realised that you also have to build https://github.com/raspberrypi/picotool/tree/add-boot2-info in order to get `picotool` to display this new info.

```
$ ~/picotool/build.add-boot2-info/picotool info -a test/kitchen_sink/kitchen_sink.uf2
File test/kitchen_sink/kitchen_sink.uf2:

Program Information
 name:            Wombat tentacles
 features:        UART stdin / stdout
 binary start:    0x10000000
 binary end:      0x10005f48
 embedded drive:  0x00080000-0x000c0000 (256K): foo

Fixed Pin Information
 0:  UART0 TX
 1:  UART0 RX

Build Information
 sdk version:       1.0.1
 pico_board:        pico
 boot2_name:        boot2_w25q080
 build date:        Feb 19 2021
 build attributes:  Release
```

(could / should there be some way for `picotool info -a` to dump out all binary_info, even that which it doesn't specifically recognise? :man_shrugging: )

File: src/rp2_common/pico_standard_link/binary_info.c [hunk 1]
```diff
@@ -66,7 +66,13 @@ bi_decl(bi_program_url(PICO_PROGRAM_URL))
 #endif
 #endif
 
-#if !PICO_NO_BUILD_TYPE_FEATURE
```
Comment (Wren6991) [inline src/rp2_common/pico_standard_link/binary_info.c:69]: Had to go looking to see where PICO_NO_BUILD_TYPE_FEATURE existed, seems this was a historic typo? Nice

This does raise that there are no PICO_CONFIG metacomments for the bininfo defines. Want me to add those?

**Review by Wren6991 — APPROVED** [review]
Works great, also on my projects with custom boot2s.

File: src/rp2_common/pico_standard_link/binary_info.c (hunk 1 above)
Comment (kilograham) [reply src/rp2_common/pico_standard_link/binary_info.c:69]: sure

**Pull request merged**